R/tar_flood_exposure.R

```diff
@@ -93,27 +113,30 @@ zonal_pop_exposure <- function(floodscan_path=fp_fs,
         }
         ) %>%
         list_rbind()
-    }
-    
   }
   return(ret)
-
 }
 
-
-
 #' floodscan_lookup
 #' @description
 #' helper function used within `zonal_pop_exposure()` to help organize, group, and aggregate FloodScan raster
 #' 
 #' @param r_fs `spatRaster` object returned from fs_to_raster() (floodScan to raster)
-#'
+#' @param country_seasons `character vector` seasons for each country
+#' 
 #' @return `tibble` lookup table with dates and months of floodscan data to be used to group and summarise FloodScan raster
 
-floodscan_lookup <-  function(r_fs){
```

Comment:
I think it would be easier if we just end the floodscan lookup after mutating `fs_year`

```r
floodscan_lookup <-  function(r_fs){
  # make a lookup table to be used for raster manipulations
  fs_mos<- floor_date(as_date(names(r_fs)),"month")

 tibble(
    fs_name = as_date(names(r_fs))
  ) |>
  mutate(
      fs_mo = floor_date(fs_name,"month"),
      fs_yr = floor_date(fs_name,"year")
    )
}
```

The additional processing for custom seasons can be added on as separate steps/functions. That way this function will remain useful across projects. You will see this is what I did in the linked repo I shared above